### DISCUSSION TITLE
[Q] salesforce static resource - unable to load the javascript library exceljs > 1.15.0 as static resource

### DISCUSSION THREAD

**Discussion by headoftheport** [exceljs/exceljs#2426] [Q&A]
I have trying import the javascript library "**exceljs**" as static resource in my lightning web component using "**loadScript**" method, the exceljs version 1.15.0 is working fine. But when I add any version higher than 1.15 the load method is unable to import the library, can the error value in the catch block of the loadScript promise is coming as "**undefined**". Please help me understand what are the possible reason for this behaviour and provide a way to use the latest version of the exceljs as the static resource.

Source:
[Version 1.15.0][1]
[Version 4.1.1][2]

  [1]: https://cdn.jsdelivr.net/npm/exceljs@1.15.0/dist/exceljs.min.js
  [2]: https://cdn.jsdelivr.net/npm/exceljs@4.1.1/dist/exceljs.min.js

**Comment by kardashovdmitry** [comment]
Hello @headoftheport,
Did you resolve the problem? I have the same issue and see the 'regeneratorRuntime is not defined [regeneratorRuntime is not defined]' message in debug mode

**Comment by ValerioSevilla** [comment]
Hi,

We found the same issue when trying to load the library as a static resource in Lightning Web Components, and also, in Aura components, library was not loading. This seems to be related to [this line of the regenerator-runtime source code](https://github.com/facebook/regenerator/blob/main/packages/runtime/runtime.js#L737). We have tested generating the ExcelJS bundle (with `npm run build`) and then "hacking" it by modifying that line:

```javascript
window.regeneratorRuntime = runtime;
```

In our case, we have both strict mode and CSP. According to the comments right below that line, a possible solution may be running that code without the strict mode. This would involve changing the ExcelJS bundler somehow.
